medusa/providers/torrent/json/t411.py

```diff
@@ -119,13 +119,15 @@ def parse(self, data, mode):
         """
         items = []
 
-        unsorted_torrent_rows = data.get('torrents') if mode != 'RSS' else data
-
-        if not unsorted_torrent_rows or not isinstance(unsorted_torrent_rows, dict):
+        # Data is `dict` when episode search and `list` when RSS search
+        if not data or not ((isinstance(data, dict) and data.get('torrents')) or
+                            isinstance(data, list) and len(data) > 0):
```

Comment:
This code is to check if we have results. Not if we have data. You missed the point.